### PR TITLE
Fix Issue 22121 -  [REG 2.097][ICE] Segmentation fault in in dmd.dsymbol.ScopeDsymbol.addAccessiblePackage

### PR DESCRIPTION
**Pull request by RazvanN7**
cc @ibuclaw

### CODE REVIEW

**Comment by RazvanN7** [comment]
Seems like the bot is down.

**Comment by kinke** [comment]
FWIW, the copypasta didn't include https://github.com/dlang/dmd/blob/bed06f4799955761eb4ac3c30bb0e1d7c250c159/src/dmd/dimport.d#L259-L264 This PR adds it.

**Review by kinke — APPROVED** [review]

**Review by ibuclaw — APPROVED** [review]
OK, but there's still one more regression maybe related to this (I'll move it to another issue).

File: test/fail_compilation/fail22121.d [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// https://issues.dlang.org/show_bug.cgi?id=22121
+
```
Comment (ibuclaw) [inline test/fail_compilation/fail22121.d:2]:
```suggestion
EXTRA_FILES: imports/test22121/package.d
```

File: test/fail_compilation/fail22121.d [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+/*
+TEST_OUTPUT:
+---
+fail_compilation/imports/test22121/package.d(1): Error: package name 'fail_compilation' conflicts with usage as a module name in file fail_compilation/fail22121.d
+---
+*/
+
+module fail_compilation;
```
Comment (ibuclaw) [inline test/fail_compilation/fail22121.d:10]: Maybe just call this
```suggestion
module fail22121;
```
then...

File: test/fail_compilation/fail22121.d [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+import fail_compilation.imports.test22121;
```
Comment (ibuclaw) [inline test/fail_compilation/fail22121.d:11]: ...tweak the import to be
```suggestion
import fail22121.imports.test22121;
```

**Comment by RazvanN7** [comment]
Merging the 2 failures are unrelated.

**Pull request merged**